### ISSUE TITLE
Contradicting rules or stories found

### ISSUE THREAD

**Issue by pheel** [RasaHQ/rasa#7301]
Sending the following e2e file off to regular training

```yaml
rules:
  - rule: Thank you
    steps:
      - intent: thankyou
        user: thanks!!!
      - action: utter_noworries
```

causes a **Contradicting rules or stories found** error. Using the same files for training and testing is a convenience. I suggest to ignore the `user` key when e2e mode if off.

**Rasa version**: 2.0.x

**Comment by sara-tagger** [comment]
Thanks for the issue, @TyDunn will get back to you about it soon!

###### You may find help in the [docs](https://rasa.com/docs/) and the [forum,](https://forum.rasa.com/) too 🤗

**Comment by samsucik** [comment]
@pheel have you checked that there are actually no contradicting rule or story parts in your training data? What exactly does the error say and what do your training stories/rules look like?

I should also mention that there have been some bugs uncovered recently in `RulePolicy` that were causing this kind of error to be raised when using a slot with `initial_value` set while also using a rule with `conversation_start: true` (see #7235 ), but that might not affect you of course :-)

**Comment by pheel** [comment]
@samsucik Thanks for getting back to me! Yes I did come upon that PR when researching, but it doesn't seem related. Here's the config and error. Removing the `user` key in the intent step makes it train fine.

**data.yml**
```yaml
rules:
  - rule: Thank you
    steps:
      - intent: thankyou
        user: thanks!!!
      - action: utter_noworries
```

**domain.yml**
```yaml
intents:
  - thankyou

responses:
  utter_noworries:
  - text: no worries!
```

**config.yml**
```yaml
...
policies:
  - name: RulePolicy
  - name: TEDPolicy
```

**error**
```
InvalidRule:
Contradicting rules or stories found 🚨

- the prediction of the action 'utter_greet' in rule 'Thank you' is contradicting with rule(s) ''.
- the prediction of the action 'action_listen' in rule 'Thank you' is contradicting with rule(s) ''.
Please update your stories and rules so that they don't contradict each other.
You can find more information about the usage of rules at https://rasa.com/docs/rasa/rules.
```

**Comment by samsucik** [comment]
Thanks a lot @pheel! I've managed to reproduce it. Will dig deeper into it. While the `user` key looks weird to be included in a rule, it's also true that rules and stories shouldn't really differ -- and in this case including the key breaks things when in a rule but doesn't when in a training story...

**Comment by samsucik** [comment]
Hmmm, right, thanks a lot for providing the details @pheel. I was able to reproduce it and will take a closer look. While it could be fixed the way you've proposed in your PR, I want to look into `RulePolicy` as well to see why this triggers such an unrelated error at all.

**Comment by samsucik** [comment]
Leaving this open as it should be addressed in #7360.

**Comment by wochinge** [comment]
closed by https://github.com/RasaHQ/rasa/issues/7360
